src/controller/CHIPDeviceController.cpp

```diff
@@ -2552,12 +2552,12 @@ CHIP_ERROR DeviceCommissioner::ParseNetworkCommissioningInfo(ReadCommissioningIn
                 ChipLogProgress(Controller, "NetworkCommissioning Features: has WiFi. endpointid = %u", path.mEndpointId);
                 info.network.wifi.endpoint = path.mEndpointId;
             }
-            else if (features.Has(NetworkCommissioning::Feature::kThreadNetworkInterface))
+            if (features.Has(NetworkCommissioning::Feature::kThreadNetworkInterface))
```

Comment:
reading through the spec, I see that data qualities are:

```
"O.a" means exactly one of the "a" elements SHALL be supported.
```

and network commissionig features sees them as:
 
 
<img width="1537" height="611" alt="image" src="https://github.com/user-attachments/assets/6a1a760b-5e35-4893-9eb2-22ca9f3f1bb2" />


Does this not mean that network commissioning MUST only support exactly one of these features and not multiple (so removing the else should not be needed)? 

This seems to be throughout the implementation, where existence of say thread will not allow wifi features to be enabled, like https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/network-commissioning/NetworkCommissioningCluster.cpp#L175

Reply:
The device should probably in that case expose the network comissioning on separate endpoints, since they are independent.